ansible_runner/loader.py

```diff
@@ -0,0 +1,175 @@
+
+from yaml import safe_load, YAMLError
+
+from ansible_runner.exceptions import ConfigurationError
+
+
+class ArtifactLoader(object):
+    '''
+    Handles loading and caching file contents from disk
+
+    This class will load the file contents and attempt to deserialize the
+    contents as either JSON or YAML.  If the file contents cannot be
+    deserialized, the contents will be returned to the caller as a string.
+
+    The deserialized file contents are stored as a cached object in the
+    instance to avoid any additional reads from disk for subsequent calls
+    to load the same file.
+    '''
+
+    logger = logging.getLogger('ansible-runner')
+
+    def __init__(self, base_path):
+        self._cache = {}
+        self.base_path = base_path
+
+    def _load_json(self, contents):
+        '''
+        Attempts to deserialize the contents of a JSON object
+
+        Args:
+            contents (string): The contents to deserialize
+
+        Returns:
+            dict: If the contents are JSON serialized
+
+            None: if the contents are not JSON serialized
+        '''
+        try:
+            return json.loads(contents)
+        except ValueError:
+            pass
+
+    def _load_yaml(self, contents):
+        '''
+        Attempts to deserialize the contents of a YAML object
+
+        Args:
+            contents (string): The contents to deserialize
+
+        Retunrs:
+            dict: If the contents are YAML serialized
+
+            None: if the contents are not YAML serialized
+        '''
+        try:
+            return safe_load(contents)
+        except YAMLError as exc:
+            self.logger.exception(exc)
+            pass
+
+    def get_contents(self, path):
+        '''
+        Loads the contents of the file specified by path
+
+        Args:
+            path (string): The relative or absolute path to the file to
+                be loaded.  If the path is relative, then it is combined
+                with the base_path to generate a full path string
+
+        Returns:
+            string: The contents of the file as a string
+
+        Raises:
+            ConfigurationError: If the file cannot be loaded
+        '''
+        try:
+            if not os.path.exists(path):
+                raise ConfigurationError('specified path does not exist %s' % path)
+
+            with open(path) as f:
+                data = f.read()
+
+            return data
+
+        except (IOError, OSError) as exc:
+            self.logger.exception(exc)
+            raise ConfigurationError('error trying to load file contents: %s' % exc)
+
+    def abspath(self, path):
+        '''
+        Transform the path to an absolute path
+
+        Args:
+            path (string): The path to transform to an absolute path
+
+        Returns:
+            string: The absolue path to the file
+        '''
+        if path.startswith(os.path.sep) or path.startswith('~'):
+            return os.path.expanduser(path)
```

Comment:
`expanduser` could wrap the `os.path.join` on L123 since it will expand `~` if it exists or ignore it otherwise.